src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerProtectedModifierFix.cs

```diff
@@ -0,0 +1,193 @@
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Collections.Generic;
+using System.Threading;
+using System.Linq;
+
+namespace Acuminator.Analyzers.StaticAnalysis.EventHandlerModifier
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class EventHandlerProtectedModifierFix : PXCodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual.Id,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations.Id);
+
+		protected override async Task RegisterCodeFixesForDiagnosticAsync(CodeFixContext context, Diagnostic diagnostic)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var root = await context.Document
+				.GetSyntaxRootAsync(context.CancellationToken)
+				.ConfigureAwait(false);
+
+			if (root?.FindNode(context.Span) is not MethodDeclarationSyntax node)
+			{
+				return;
+			}
+
+			var isOverride = GetPropertyValue(diagnostic, "IsOverride");
+			var isExplicitInterfaceImplementation = GetPropertyValue(diagnostic, "IsExplicitInterfaceImplementation");
+			var implementsInterface = GetPropertyValue(diagnostic, "ImplementsInterface");
+			var isContainingTypeSealed = GetPropertyValue(diagnostic, "IsContainingTypeSealed");
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			if (isOverride)
+			{
+				return;
+			}
+
+			if (isExplicitInterfaceImplementation)
+			{
+				var removeExplicitInterface = nameof(Resources.PX1077Fix_RemoveExplicitInterface).GetLocalized().ToString();
+				var codeFixAction = new RemoveExplicitInterfaceAction(removeExplicitInterface, context.Document, node);
+
+				context.RegisterCodeFix(codeFixAction, diagnostic);
+			}
+			else
+			{
+				if (!implementsInterface)
+				{
+					var accessibilityModifier = isContainingTypeSealed
+						? SyntaxKind.PublicKeyword
+						: SyntaxKind.ProtectedKeyword;
+
+					var addVirtualModifier = !isContainingTypeSealed;
+
+					var localizedText = SyntaxFactory.Token(accessibilityModifier).Text;
+					if (addVirtualModifier)
+					{
+						localizedText += " " + SyntaxFactory.Token(SyntaxKind.VirtualKeyword).Text;
+					}
+
+					var makeProtectedTitle = nameof(Resources.PX1077Fix).GetLocalized(localizedText).ToString();
+					var codeFixAction = new ChangeAccessibilityModifierAction(makeProtectedTitle, context.Document, node, accessibilityModifier, addVirtualModifier);
+
+					context.RegisterCodeFix(codeFixAction, diagnostic);
+				}
+			}
+		}
+
+		private static bool GetPropertyValue(Diagnostic diagnostic, string propertyName)
+		{
+			if (diagnostic.TryGetPropertyValue(propertyName, out string? strValue) && bool.TryParse(strValue, out bool value))
```

Comment:
There is a helper extension method `IsFlagSet` that already does all this:
https://github.com/Acumatica/Acuminator/blob/dev/src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DiagnosticUtils.cs#L22

Reply:
Done.